File: AccelerationToMulitcore/1.CPUOpenmp/04.Extras/05tasks/openmptask0.f90
program fibonacci

        implicit none

        integer :: x, y
        x = 10
        y = fib(x)

        print*, y

        contains

        RECURSIVE INTEGER FUNCTION fib(n) RESULT(res)
                INTEGER n, i, j
                IF ( n .LT. 2) THEN
                        res = n
                ELSE
              !$OMP TASK SHARED(i)
                 i = fib( n-1 )
              !$OMP END TASK

               !$OMP TASK SHARED(j)
                 j = fib( n-2 )
              !$OMP END TASK

              !$OMP TASKWAIT
                res = i+j
              END IF
        end function


end program fibonacci

File: AccelerationToMulitcore/1.CPUOpenmp/04.Extras/05tasks/openmptask0.c
#include <stdio.h>
#include <omp.h>
int fib(int n)
{
	if (n<2) return n;
	else
	{
	int x, y;

	#pragma omp task shared(x) if(n>30) 
	x = fib(n-1); 

	#pragma omp task shared(y) if(n>30) 
	y = fib(n-2);

	#pragma omp taskwait	
	return x+y; 
	}
} 


int main()
{
  int n = 10;

  omp_set_dynamic(0);
  omp_set_num_threads(4);

  #pragma omp parallel
  {
    #pragma omp master
    printf ("fib(%d) = %d\n", n, fib(n));
  }
}
